python/ray/tests/test_reference_counting.py

```diff
@@ -236,15 +258,296 @@ def get_large_object(self):
     actor = Actor.remote()
     ray.get(actor.wait_for_actor_to_start.remote())
 
-    for batch in range(10):
-        intermediate_result = f.remote(
-            np.zeros(10 * 1024 * 1024, dtype=np.uint8))
-        ray.get(intermediate_result)
-
     # The ray.get below fails with only LRU eviction, as the object
     # that was ray.put by the actor should have been evicted.
-    with pytest.raises(ray.exceptions.RayTimeoutError):
-        ray.get(actor.get_large_object.remote(), timeout=1)
+    _fill_object_store_and_get(actor.get_large_object.remote(), succeed=False)
+
+
+# Remote function takes serialized reference and doesn't hold onto it after
+# finishing. Referenced object shouldn't be evicted while the task is pending
+# and should be evicted after it returns.
+@pytest.mark.skip("Serialized ObjectID reference counting not implemented.")
+def test_basic_serialized_reference(one_worker_100MiB):
+    @ray.remote
+    def pending(ref, dep):
+        ray.get(ref[0])
+
+    # TODO(edoakes): currently these tests don't work with ray.put() so we need
+    # to return from a task like this instead. Once that is fixed, should have
+    # tests run with both codepaths.
+    @ray.remote
+    def put():
+        return np.zeros(40 * 1024 * 1024, dtype=np.uint8)
```

Comment:
Do you want to also add tests for testing task returns vs ray.put()?

Reply:
I'll add that with the PR that fixes ray.put